samples/cellular/modem_shell/src/sock/sock_shell.c

```diff
@@ -212,6 +229,100 @@ static void sock_print_usage(enum sock_shell_command command)
 	}
 }
 
+static int cmd_sock_getaddrinfo(const struct shell *shell, size_t argc, char **argv)
+{
+	int err = 0;
+
+	if (argc < 2) {
+		goto show_usage;
+	}
+
+	/* Variables for command line arguments */
+	int arg_family = AF_UNSPEC;
+	int arg_type = 0;
+	char arg_hostname[SOCK_MAX_ADDR_LEN + 1];
+	int arg_pdn_cid = 0;
+
+	memset(arg_hostname, 0, SOCK_MAX_ADDR_LEN + 1);
+
+	optreset = 1;
+	optind = 1;
+	int opt;
+
+	while ((opt = getopt_long(argc, argv, short_options, long_options, NULL)) != -1) {
+		int addr_len = 0;
+
+		switch (opt) {
+		case 'H': /* Hostname */
+			addr_len = strlen(optarg);
+			if (addr_len > SOCK_MAX_ADDR_LEN) {
+				mosh_error(
+					"Hostname length %d exceeded. Maximum is %d.",
+					addr_len, SOCK_MAX_ADDR_LEN);
+				return -EINVAL;
+			}
+			memcpy(arg_hostname, optarg, addr_len);
+			break;
+		case 'f': /* Address family */
+			if (strcmp(optarg, "unspec") == 0) {
+				arg_family = AF_UNSPEC;
+			} else if (strcmp(optarg, "inet") == 0) {
+				arg_family = AF_INET;
+			} else if (!strcmp(optarg, "inet6")) {
+				arg_family = AF_INET6;
+			} else {
+				mosh_error(
+					"Unsupported address family=%s. Supported values are: "
+					"'unspec', 'inet' (ipv4) or 'inet6' (ipv6)",
+					optarg);
+				return -EINVAL;
+			}
+			break;
+		case 't': /* Socket type */
+			if (strcmp(optarg, "any") == 0) {
+				arg_type = 0;
+			} else if (strcmp(optarg, "stream") == 0) {
+				arg_type = SOCK_STREAM;
+			} else if (strcmp(optarg, "dgram") == 0) {
+				arg_type = SOCK_DGRAM;
+			} else {
+				mosh_error(
+					"Unsupported address type=%s. Supported values are: "
+					"'any', 'stream' (tcp) or 'dgram' (udp)",
+					optarg);
+				return -EINVAL;
+			}
+			break;
+		case 'I': /* PDN CID */
+			arg_pdn_cid = atoi(optarg);
+			if (arg_pdn_cid < 0) {
+				mosh_error("PDN CID (%d) must be positive integer.", arg_pdn_cid);
```

Comment:
If you want to be precise, the error text should also be updated. Zero is not positive.

Reply:
Actually I will change this check to be a range `0-10` according to the [AT+CGDCONT](https://docs.nordicsemi.com/bundle/ref_at_commands_nrf91x1/page/REF/at_commands/packet_domain/cgdcont_read.html) documentation.

Reply:
I wouldn't limit the number to 10 because this is a test tool where we can try to allow use of some invalid input in order to test modem behavior.
We don't check all validities due to this but I'm sure it's not consistent throughout the application.

Reply:
If we want to be able to test any value I think we should remove the validation and accept anything into `arg_pdn_cid`.

Reply:
I think removing the validation is better than checking whether it's higher than 10.
I don't have strong opinion about checking negative values so I'm fine keeping it or removing it. There are few other places in MOSH that does some checks for CID but we can handle those later.

Reply:
As I said, I'd remove the check about being bigger than 10. And for negative values the check or no check is fine for me.

Reply:
I think that something like this would be acceptable:
```suggestion
				mosh_error("PDN CID (%d) must be non-negative integer.", arg_pdn_cid);
```

Reply:
ok 👍